bindings/nodejs/src/functions/uart.cc

```diff
@@ -140,36 +168,38 @@ NAN_METHOD(bind_sol_uart_write) {
         return;
 
     sol_uart *uart = uart_data->uart;
-    unsigned char *outputBuffer = (unsigned char *) 0;
-
     if (!node::Buffer::HasInstance(info[1])) {
         Nan::ThrowTypeError("Argument 1 must be a Buffer");
         return;
     }
 
+    struct sol_blob *blob;
     size_t length = node::Buffer::Length(info[1]);
-
-    outputBuffer = (unsigned char *) malloc(length * sizeof(unsigned char));
+    void *outputBuffer = malloc(length);
     if (!outputBuffer) {
         Nan::ThrowError("Failed to allocate memory for output buffer");
         return;
     }
 
     memcpy(outputBuffer, node::Buffer::Data(info[1]), length);
+    blob = sol_blob_new(&SOL_BLOB_TYPE_DEFAULT, NULL, outputBuffer, length);
```

Comment:
Here is where the new pattern and used types will shine by reducing copies and memory usage.

The purpose of `struct sol_buffer` is to allow users to define read-only memory that has reference counting and parenting, as well as a custom type that provides its own "free".

That said, if you hava JS object you should create a blob for it, increase its reference count (since the blob will keep it), and when the blob is deleted, you unref the JS object. You can do this using two ways, pick whatever is the best:
 1. create a blob type for `SOL_BLOB_TYPE_JS_OBJECT`, use `data = object, len = whatever`, on `SOL_BLOB_TYPE_JS_OBJECT->free()` you unref `data`. In this case, you create a second blob of type `SOL_BLOB_TYPE_NO_FREE_DATA` for the actual buffer, `data = buffer, len = buffer_length, parent = jsobject_blob`. Then you feed this child blob. It will take a reference to the parent, when it not used anymore, the parent will be freed and the JS Object will be unreferenced;
 2. create a blob type for `SOL_BLOB_TYPE_JS_BUFFER` with a new extension/subclass structure: `struct sol_blob_js_buffer { struct sol_blob base; JSBuffer *js_buffer; }`. In this case you use `base.data = buffer, base.len = buffer_length; js_buffer = js_buffer->ref()`. Your new `SOL_BLOB_TYPE_JS_BUFFER->free()` will only `blob->js_buffer->unref(); free(blob);`

They are marginally different and provide the same net effect: no copies of your JS buffer. In your current code it would save 2 copies, if you imagine a huge buffer it would make a big difference :-)

The first approach I've mentioned is easier to extend to different objects, they could all use the same parent type. However it will allocate one more `struct sol_blob`. The second approach works only for buffer, but is slightly more efficient. Since the expected users are to be "buffer" kind, I don't see a problem going with the second approach.

IPM was recently introduced using this pattern, see example code at https://github.com/solettaproject/soletta/blob/master/src/lib/io/sol-ipm.c#L59 (ignore what is IPM, just see the blob declared in this line)